src/main/java/org/gridsuite/directory/server/DirectoryController.java

```diff
@@ -86,10 +87,12 @@ public ResponseEntity<Flux<ElementAttributes>> getRootDirectories(@RequestHeader
     @Operation(summary = "Get if a root directory of this name exists")
     @ApiResponses(value = {
         @ApiResponse(responseCode = "200", description = "The root directory exists"),
-        @ApiResponse(responseCode = "404", description = "The root directory doesn't exist"),
+        @ApiResponse(responseCode = "204", description = "The root directory doesn't exist"),
     })
     public ResponseEntity<Mono<Void>> rootDirectoryExists(@RequestParam("directoryName") String directoryName) {
-        return ResponseEntity.ok().body(service.rootDirectoryExists(directoryName));
+        return service.rootDirectoryExists(directoryName) ?
```

Comment:
May be factorize out what is common ?
```suggestion
        HttpStatus status = service.rootDirectoryExists(directoryName) ? HttpStatus.OK : HttpStatus.NO_CONTENT;
        return ResponseEntity.status(status)
                    .contentType(MediaType.APPLICATION_JSON).body(Mono.empty());

```

Reply:
Yes, that's better.